x/gamm/keeper/pool_service_test.go

```diff
@@ -891,32 +910,34 @@ func (suite *KeeperTestSuite) TestJoinSwapExactAmountInConsistency() {
 					ExitFee: tc.poolExitFee,
 				},
 			)
+			if err != nil {
+				suite.Require().Equal(err, tc.expectError)
```

Comment:
I personally think that the expected error in unit tests should reflect an error that might happen inside a function we are testing, which is, in this case, `JoinSwapExactAmountIn`. I think if we want to test edge cases for balancer pool creation, it should be done in a separate test

Reply:
if, for any reason, something breaks for pool creation, when the expected error is related to `JoinSwapExactAmountIn`, this require statement will compare an error happened during pool creation with some internal to `JoinSwapExactAmountIn` edge case